tools/src/bin/output-file.rs

```diff
@@ -126,7 +128,46 @@ fn main() {
             }
         }
 
-        let panel = if path.contains("__GENERATED__") {
+        let mut extension_mapping = HashMap::new();
+        extension_mapping.insert("cpp", ("header", vec!["h", "hh"]));
+        extension_mapping.insert("cc", ("source", vec!["h", "hh"]));
+        extension_mapping.insert("cxx", ("source", vec!["h", "hh"]));
+        extension_mapping.insert("h", ("source", vec!["cpp", "cc", "cxx"]));
+
+        let extension = path_wrapper.extension().unwrap_or(&OsStr::new("")).to_str().unwrap();
+        let show_header = match extension_mapping.get(extension) {
+            Some(&(ref description, ref try_extensions)) => {
+                let mut result = None;
+                for try_ext in try_extensions {
+                    let try_buf = path_wrapper.with_extension(try_ext);
+                    let try_path = try_buf.as_path();
+                    if try_path.exists() {
+                        let (path_sec, _) = path.split_at(path.len() - extension.len() - 1);
```

Comment:
Not clear to me what _sec stands for. I would have called it path_dir or path_base or something.